### DISCUSSION TITLE
Get cost function value at the each simulation step

### DISCUSSION THREAD

**Discussion by AleksandrKashirin** [do-mpc/do-mpc#347] [Q&A]
Hello, dear do_mpc community!
Is there any way to get cost function value at the each simulation step? I want to understand the correctness of my work and it will be much easier with this knowledge.

- Aleksandr

**Comment by 4flixt** [comment]
Hi @AleksandrKashirin ,

this is definitely a good idea to diagnose your MPC controller. You can use ``model.set_expression`` to define your cost in the  model class. This cost is then available for inspection. We use this trick in this [example](https://www.do-mpc.com/en/latest/example_gallery/oscillating_masses_discrete.html).

Best,
4flixt
